src/common/mempool.cc

```diff
@@ -100,7 +100,7 @@ void mempool::pool_t::get_stats(
     total->bytes += shard[i].bytes;
   }
   if (debug_mode) {
-    std::unique_lock<std::mutex> shard_lock(lock);
+    std::lock_guard<std::mutex> shard_lock(lock);
```

Comment:
no need to specify the template parameter, C++17 supports class template argument deduction.

Reply:
I'm not sure removing such parameters is better style, in this case.

Matt

Reply:
i think the general goal is to avoid unnecessary repeating of information. it's quite the same as what we want from `make_pair()`.